tox.ini

```diff
@@ -71,10 +71,11 @@ deps =
     --requirement={toxinidir}/requirements-dev.txt
     --requirement={toxinidir}/yelp_package/extra_requirements_yelp.txt
     --editable={toxinidir}
+whitelist_externals = bash
 commands =
     check-requirements -vv
     pre-commit install -f --install-hooks
-    pre-commit run --all-files
+    bash -c 'pre-commit run --all-files || (if [[ $HOSTNAME =~ dev.* ]]; then echo "Re-running pre-commit since this is a dev box"; pre-commit run --all-files; else false; fi)'
```

Comment:
just curious: what is this for? is this to handle hooks that might want to re-run on the fixed output from the first run?

Reply:
Yeah, exactly. I'm tired of having to rerun it myself when e.g. black fixes things automatically. This will still fail if the hooks can't fix something.